sources/packages/backend/libs/integrations/src/services/disbursement-schedule/e-cert-calculation/e-cert-calculation-process.ts

```diff
@@ -152,13 +170,56 @@ export abstract class ECertCalculationProcess {
     }
   }
 
+  /**
+   * Checks whether the disbursement blocked notification should be created or not.
+   * @param disbursementId disbursement id.
+   * @param entityManager entity manager to execute in transaction.
+   * @returns boolean indicating if the disbursement blocked notification should be created or not.
+   */
+  private async shouldCreateDisbursementBlockedNotification(
+    disbursementId: number,
+    entityManager: EntityManager,
+  ): Promise<boolean> {
+    // Create notifications to the student and ministry when the disbursement is blocked.
+    const notificationMessageId = 16;
+    const notification = await entityManager
+      .getRepository(Notification)
+      .createQueryBuilder("notifications")
+      .select("count(*)::int", "notificationsCount")
+      .addSelect("max(notifications.created_at)", "max_created_at")
+      .innerJoin("notifications.notificationMessage", "notificationMessage")
+      .where("notificationMessage.id = :notificationMessageId", {
+        notificationMessageId,
+      })
+      .andWhere("notifications.permanent_failure_error IS NULL")
+      .andWhere(
+        "notifications.metadata->>'disbursement_id' = :disbursementId",
+        {
+          disbursementId,
+        },
+      )
+      .getRawOne<NotificationData>();
+    const sevenDaysLaterDate = getUTC(addDays(7, notification.max_created_at));
```

Comment:
Can we have a constant for this "7"?

Reply:
Also, instead of `sevenDaysLaterDate` find another more meaningful name. 
"Code logic updated to store the metadata informationin db and to ensure the notifications are not sent after 3 emails and are sent every 7 days."
I think we could call it something like `minimumDateToSendANewMessage`.

Reply:
@sh16011993 the constant can be created in the below file:
libs\services\src\constants\system-configurations-constants.ts

Reply:
Moved it there.